Libs/vtkITK/vtkITKArchetypeImageSeriesScalarReader.cxx

```diff
@@ -114,87 +113,87 @@ int vtkITKArchetypeImageSeriesScalarReader::RequestData(
 
 /// SCALAR MACRO
 #define vtkITKExecuteDataFromSeries(typeN, type) \
-    case typeN: \
-    {\
-      typedef itk::Image<type,3> image##typeN;\
-      typedef itk::ImageSource<image##typeN> FilterType; \
-      FilterType::Pointer filter; \
-      itk::ImageSeriesReader<image##typeN>::Pointer reader##typeN = \
-        itk::ImageSeriesReader<image##typeN>::New(); \
-      vtkITKExecuteDataDeclareDICOMImageIO \
-      if (this->ArchetypeIsDICOM) \
-      { \
-        reader##typeN->SetImageIO(imageIO); \
-      } \
-      itk::CStyleCommand::Pointer pcl=itk::CStyleCommand::New(); \
-      pcl->SetCallback((itk::CStyleCommand::FunctionPointer)&ReadProgressCallback); \
-      pcl->SetClientData(this); \
-      reader##typeN->AddObserver(itk::ProgressEvent(),pcl); \
-      reader##typeN->SetFileNames(this->FileNames); \
-      reader##typeN->ReleaseDataFlagOn(); \
-      if (this->UseNativeCoordinateOrientation) \
-      { \
-        filter = reader##typeN; \
-      } \
-      else \
-      { \
-        itk::OrientImageFilter<image##typeN,image##typeN>::Pointer orient##typeN = \
-            itk::OrientImageFilter<image##typeN,image##typeN>::New(); \
-        if (this->Debug) {orient##typeN->DebugOn();} \
-        orient##typeN->SetInput(reader##typeN->GetOutput()); \
-        orient##typeN->UseImageDirectionOn(); \
-        orient##typeN->SetDesiredCoordinateOrientation(this->DesiredCoordinateOrientation); \
-        filter = orient##typeN; \
-      }\
-      filter->UpdateLargestPossibleRegion(); \
-      itk::ImportImageContainer<itk::SizeValueType, type>::Pointer PixelContainer##typeN;\
-      PixelContainer##typeN = filter->GetOutput()->GetPixelContainer();\
-      void *ptr = static_cast<void *> (PixelContainer##typeN->GetBufferPointer());\
-      DownCast<type>(data->GetPointData()->GetScalars())                \
-        ->SetVoidArray(ptr, PixelContainer##typeN->Size(), 0,\
-                       vtkAOSDataArrayTemplate<type>::VTK_DATA_ARRAY_DELETE);\
-      PixelContainer##typeN->ContainerManageMemoryOff();\
+  case typeN: \
```

Comment:
I know it is not a regression and does not have to be addressed in this pull request, but such long macros are terrible (hard to read, impossible to debug). It should be replaced in a follow-up pull request with a short macro (if needed) and a templated function.

Reply:
It is ugly indeed.